docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,218 @@
+# CSS variables
+
+<p class="description">Learn about the experimental API for using CSS variables with Material UI components.</p>
+
+CSS variables provide significant improvements in developer experience related to theming and customization.
+With these variables, you can inject a theme into your app's stylesheet _at build time_ to apply the user's selected settings before the whole app is rendered.
+
+This solves the problem of dark-mode SSR flickering; lets you provide your users with multiple themes beyond light and dark; and offers a better debugging experience overall, among other benefits.
+
+Previously, these CSS variables were only available as an experimental API in the MUI System package.
+Now they are ready for experimental use with Material UI components.
+
+> If you want to see wider support for this API across Material UI's component library, please feel free to contribute to the ongoing development. Make sure to check the [GitHub issue](https://github.com/mui/material-ui/issues/32049) that keeps track of our progress, to see if anyone else is currently working on a component you're interested in.
+
+## Introduction
+
+The CSS variables API relies on a new experimental provider for the theme called `Experimental_CssVarsProvider` to inject styles into Material UI components.
+In addition to providing the theme in the inner React context, this new provider also generates CSS variables out of all tokens in the theme that are not functions, and makes them available in the context as well.
+
+All of these variables are accessible in an object in the theme called `vars`.
+The structure of this object is nearly identical to the theme structure, the only difference is that the values represent CSS variables.
+
+## Usage
+
+`Experimental_CssVarsProvider` is a new experimental provider that attaches all generated CSS variables to the theme and puts them in React's context. Children elements under this provider will also be able to read the CSS variables from the theme.
+
+```js
+import { Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';
+
+function App() {
+  return <CssVarsProvider>...</CssVarsProvider>;
+}
+```
+
+### Customizing components
+
+Because the CSS variables API is an experimental feature, it is currently only supported by the `Button` component.
+To customize it using CSS variables, you'll need to wrap your application with `Experimental_CssVarsProvider`.
+
+Play around with the demo below!
+We'd appreciate any feedback about this API, as it is still in development.
```

Comment:
I think I'd remove this bit here, it's weird to be only inside the "Customizing components" sub-section.
What I'd maybe do is add a "Feedback" topic, below the "API" one, and add more instructions as to how to give feedback?! Where to open the issue, how to do it, how to structure the writing, etc.

Reply:
Yeah makes sense. It's probably better to add it before the API actually, as I think the API is the last section always on a pages where it is next to the content.

Reply:
I actually move this sentence in the top Note, feels like it belongs there, as we are already asking for help, asking for feedback can be complementary